app/src/main/java/com/github/h3lp3rs/h3lp/MedicalCardActivity.kt

```diff
@@ -49,12 +54,35 @@ class MedicalCardActivity : AppCompatActivity() {
         createBloodField()
         createGenderField()
         createPrivacyCheckBox()
+        createPhoneNumberField()
 
         createHelpField(findViewById(R.id.medicalInfoConditionTxtLayout), getString(R.string.condition_help_msg))
         createHelpField(findViewById(R.id.medicalInfoTreatmentTxtLayout), getString(R.string.treatment_help_msg))
         createHelpField(findViewById(R.id.medicalInfoAllergyTxtLayout), getString(R.string.allergy_help_msg))
     }
 
+
+    /**
+     * initialises the field for emergency contact phone number so that we make
+     * sure to allow only valid phone numbers, or display an error.
+     */
+    private fun createPhoneNumberField() {
+        val phoneInputText = findViewById<EditText>(R.id.medicalInfoContactNumberEditTxt)
+        val phoneInputLayout = findViewById<TextInputLayout>(R.id.medicalInfoContactNumberTxtLayout)
+        phoneInputText.doOnTextChanged { text, _, _, _ ->
+            try {
+                val number = PhoneNumberUtil.getInstance().parse(text.toString().trimStart('0'), "CH")
```

Comment:
This should be made dependent of the location we are in.  I know this is not something that should be done for this sprint, but it should be done later. Maybe add a TODO here to remind ourselves of this.

Reply:
The "CH" may look hardcoded but is in fact the default value in case the parder cannot infer the country prefix, thus it might be overkill to make it location dependant, but it's true that we could do it, possibly by enhancing the country emergency numbers database. Anyway adding a comment seems a must :)